### ISSUE TITLE
pandevice.errors.PanDeviceXapiError: Illegal parameter [request]

### ISSUE THREAD

**Issue by irom77** [PaloAltoNetworks/pandevice#13]
I tried to display job status on Panorama:

```
$ cat pandevice.dev-pano.commit_all.py
#!/usr/bin/python
from pandevice import base
from pandevice import firewall
from pandevice import panorama
from pandevice import policies
from pandevice import objects
from pandevice import network
from pandevice import device
pano = panorama.Panorama("xxx", "admin", "xxx")
print pano.op("show jobs id 4936", xml=True)
```

and got below error (using pandevice 0.3.5 and python 2.7.5)

```
Traceback (most recent call last):
  File "pandevice.dev-pano.commit_all.py", line 16, in <module>
    print pano.op("show jobs id 4936", xml=True)
  File "/usr/lib/python2.7/site-packages/pandevice/panorama.py", line 137, in op
    return super(Panorama, self).op(cmd, vsys=None, xml=xml, cmd_xml=cmd_xml, extra_qs=extra_qs, retry_on_peer=retry_on_peer)
  File "/usr/lib/python2.7/site-packages/pandevice/base.py", line 1664, in op
    element = self.xapi.op(cmd, vsys, cmd_xml, extra_qs, retry_on_peer=retry_on_peer)
  File "/usr/lib/python2.7/site-packages/pandevice/base.py", line 1573, in method
    raise the_exception
pandevice.errors.PanDeviceXapiError: Illegal parameter [request]
```

**Comment by btorresgil** [comment]
Thanks for the details of the problem.  A PanDeviceXapiError typically indicates an error message coming from the device as the result of an API call.  In this case, the Panorama is replying with a message "Illegal parameter".

The cause of this error is actually due to how the command is formed when you call the .op() method.  To convert the command correctly for the API call, the job id must be in double quotations.  For example:

    # notice the double quotes around 4936
    print pano.op('show jobs id "4936"', xml=True)

The reason the quotation marks are needed is explained in the pan-python documentation (pan-python is a dependency of pandevice):
https://github.com/kevinsteves/pan-python/blob/master/doc/pan.xapi.rst#opcmdnone-vsysnone-cmd_xmlfalse

In that doc, it explains that the command is converted to XML before it gets passed to the device.  Parts of the command XML must be XML tags, while other parts are text within tags.  Anything inside double quotes will be treated as text within a tag, so the correctly converted command XML should look like this:

`<show><jobs><id>4936</id></jobs></show>`

Optionally you can pass the XML in directly, bypassing conversion to XML, by setting cmd_xml to False.  For example:

    # this also works
    print pano.op('<show><jobs><id>4936</id></jobs></show>', xml=True, cmd_xml=False)

That way the library is expecting XML and won't attempt the conversion to XML before making the API call to the device.

**tl;dr:**   In summary, the easiest solution is to change the method call by putting double quotations around the job id:

    print pano.op('show jobs id "4936"', xml=True)
